packages/backend/src/index.ts

```diff
@@ -80,16 +82,26 @@ async function main() {
   const graphqlEnv = useHotMemoize(module, () => createEnv('graphql'));
   const appEnv = useHotMemoize(module, () => createEnv('app'));
 
+  passport.use(
+    new BackstageIdentityStrategy({
+      discovery: SingleHostDiscovery.fromConfig(config),
+    }),
+  );
+  const backstageAuth = passport.authenticate('backstage', {
+    session: false,
+  });
+
   const apiRouter = Router();
-  apiRouter.use('/catalog', await catalog(catalogEnv));
-  apiRouter.use('/rollbar', await rollbar(rollbarEnv));
-  apiRouter.use('/scaffolder', await scaffolder(scaffolderEnv));
+  apiRouter.use(passport.initialize());
+  apiRouter.use('/catalog', backstageAuth, await catalog(catalogEnv));
+  apiRouter.use('/rollbar', backstageAuth, await rollbar(rollbarEnv));
+  apiRouter.use('/scaffolder', backstageAuth, await scaffolder(scaffolderEnv));
   apiRouter.use('/auth', await auth(authEnv));
-  apiRouter.use('/techdocs', await techdocs(techdocsEnv));
-  apiRouter.use('/kubernetes', await kubernetes(kubernetesEnv));
-  apiRouter.use('/proxy', await proxy(proxyEnv));
-  apiRouter.use('/graphql', await graphql(graphqlEnv));
-  apiRouter.use(notFoundHandler());
+  apiRouter.use('/techdocs', backstageAuth, await techdocs(techdocsEnv));
+  apiRouter.use('/kubernetes', backstageAuth, await kubernetes(kubernetesEnv));
+  apiRouter.use('/proxy', backstageAuth, await proxy(proxyEnv));
+  apiRouter.use('/graphql', backstageAuth, await graphql(graphqlEnv));
+  apiRouter.use(backstageAuth, notFoundHandler());
```

Comment:
Does  this apply authentication for the generic 404 Handler as well ?

Reply:
Yes. This would be the adopters code, so optional. I added it as a ”do not leak info to outside hackers” thing 🤷